src/main/java/org/soen343/services/modules/SHCModule.java

```diff
@@ -78,18 +105,30 @@ public void updateLightState(HashSet<Integer> lights) {
     }
 
     public boolean setAutoMode() {
-        //Check permissions:
         Individual ind = User.getCurrentIndividual();
-        if (ind.getRole().equals("Family Adult") || ind.getRole().equals("Family Child")) {
-            Model.getSimulationParameters().setAutoMode();
-            System.out.println("[SHC Module] [Auto Mode] " + ind.getName() + " has set Auto mode to " + (Model.getSimulationParameters().isAutoModeOn() ? "ON" : "OFF"));
-            return true;
+
+        //Check permissions:
+        if (validate(0)) {
```

Comment:
What is this validate(0)?

Reply:
The validate interface takes an int id as a parameter, but the automode doesn't need it so I put a random int.